### ISSUE TITLE
How can this work for import when items in the import have FKs to other item in the import? 

### ISSUE THREAD

**Issue by mscout1** [django-import-export/django-import-export#1869]
I am working with a model that looks like this: 

```
class MenuItem(models.Model):
    caption = models.CharField(
        gettext_lazy('Tab or link name'),
        max_length=64,
        db_index=True)
    parent = models.ForeignKey(
        'self',
        on_delete=models.SET_NULL,
        blank=True,
        null=True)
```

I can export it just fine. In import it fails because the `parent` item is not yet in the database. It IS in the import. Is there a way to get this to work?

**Comment by matthewhegarty** [comment]
Please refer to [Creating non existent relations](Does [this](https://django-import-export.readthedocs.io/en/latest/advanced_usage.html#creating-non-existent-relations) help?)

**Comment by mscout1** [comment]
Using `before_import_row` and natural_keys I was able to create stubs for the parent items as I encountered the FKs. This lead to a bunch of nasty uniqueness constraint violations.

I was able to workaround it with some fairy ugly code in `before_save_instance`:

    def before_save_instance(self, instance, row, **kwargs):
        nk = instance.natural_key()
        try:
            existing = self._meta.model.objects.get_by_natural_key(*nk).pk
            instance.pk = existing
        except self._meta.model.DoesNotExist:
            pass  # Do nothing
            
Usually when you are importing something that already exists, there is nice UI about updating it. But when stubs are created on the fly, I need this workaround. I also tried using a custom widget, with much then same result. 

Was there a more elegant way to do this? 

---

A more elegant solution for my specific use case would be to sort the items so that the fie was in the correct order already. since we only have 2 levels of hierarchy this could be done be putting items with null parents first. 

The problem in that case seemed to be that the file did not respect the ordering of the query set.
